megablocks/layers/router.py

```diff
@@ -58,6 +57,7 @@ def forward(self, x):
         if self.args.moe_normalize_expert_weights:
             expert_weights = expert_weights / torch.norm(
                 expert_weights, p=self.args.moe_normalize_expert_weights,dim=-1, keepdim=True)
+        if self.args.moe_top_k == 1: expert_indices = expert_indices.squeeze(-1)
```

Comment:
I don't think this is necessary - we flatten the expert assignments immediately after the router.

Reply:
Ah, that wasn't obvious to me; now I see!